### PR TITLE
Paginate pod logs instead of truncating them

### PR DESCRIPTION
**Pull request by shtlrs**
Truncating original logs can result into important context/logs being burried/removed.

This paginates the logs instead

### CODE REVIEW

**Comment by jb3** [comment]
My main feedback on this is that without code blocks the logs become harder to read? How does this look if we move the logs into a code-block within the embeds?

**Comment by shtlrs** [comment]
> My main feedback on this is that without code blocks the logs become harder to read? How does this look if we move the logs into a code-block within the embeds?

Good point, we could do inline formatting.

Here are some examples.

Code blocks looks like this
image

Inline formatting looks like this

image


However, when we have an exception for example, code blocks becomes too bulky

image

And inline looks like this

image

**Comment by jchristgit** [comment]
```py
    @classmethod
    async def paginate(
        cls,
        lines: list[str],
        ctx: Context | discord.Interaction,
        embed: discord.Embed,
        prefix: str = "",
        suffix: str = "",
```
What about passing `prefix='```\n', suffix='\n```'` to the paginator? I feel that would heavily improve readability.

File: arthur/pagination.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        restrict_to_user: discord.User | None = None,
+        timeout: int = 300,
+        footer_text: str | None = None,
+        url: str | None = None,
+        allowed_roles: Sequence[int] | None = None,
+        *,
+        reply: bool = False,
+        empty: bool = True,
+        exception_on_empty_embed: bool = False,
+    ) -> discord.Message | None:
```
Comment (jchristgit) [inline arthur/pagination.py:37]: How come we don't make use of `*args, **kwargs` here? From my understanding we only update the emojis?

Comment (shtlrs) [reply arthur/pagination.py:37]: We only update the emojis indeed, but I prefer being explicit on what's being passed down as arguments.

**Comment by shtlrs** [comment]
> What about passing `` prefix='```\n', suffix='\n```' `` to the paginator? I feel that would heavily improve readability.

We could, and will, I just don't know which of the output samples I gave are more appreciated.

File: arthur/pagination.py (hunk 1 above)
Comment (jchristgit) [reply arthur/pagination.py:37]: My concern here is that this binds our own function's definition to `discord`s function definition, and future updates in the library may break here without us noticing.

**Comment by jchristgit** [comment]
> We could, and will, I just don't know which of the output samples I gave are more appreciated.

I like the first one more, but I would strongly prefer if all the messages were wrapped in a single codeblock, therefore the `prefix` and `suffix` suggestion.

File: arthur/pagination.py (hunk 1 above)
Comment (shtlrs) [reply arthur/pagination.py:37]: This signature hasn't changed in ages.

I see your point, but the main changes needs to happen in bot-core as the downstream's method signature doesn't allow for extra arguments.

We could do that in a separate PR if you'd like ?

**Comment by shtlrs** [comment]
> > We could, and will, I just don't know which of the output samples I gave are more appreciated.
> 
> I like the first one more, but I would strongly prefer if all the messages were wrapped in a single codeblock, therefore the `prefix` and `suffix` suggestion.

How do you like this ?

image

**Comment by shtlrs** [comment]
> I did not see this, I see this now, I think this is good, I think we should do it, what does @jb3 think?

Both Markus and Volcyy like this

**Review by jb3 — APPROVED** [review]
Looks good! Can't test but I trust you. GOAT!

**Pull request merged**